### ISSUE TITLE
Feature request: faster pasting

### ISSUE THREAD

**Issue by danielo515** [Clipy/Clipy#295]
Hello,

First of all, thanks for this great piece of open source software.
My desired usage is to just toggle between the most recent copied texts while pasting.
This is something that has been solved very elegantly by [citation needed] on windows.
This is how it works:

1. You hit the paste shortcut (usually control+v)
1. If you hold the control key, then the menu appears
1. Every time you hit the V key again while holding the control key it selects the next available item on the history
1. When you release the control key the selection is pasted

This greatly enhances my productivity because usually I want to paste the current clip or maybe one or two clips ago, and I don't want to open the clipy menu, select the history, then select the clip and then paste. Too many steps for something that I do too often.

Thanks and regards

**Comment by tiosecurity** [comment]
Bravo! I was almost ready to create the same suggestions. This is the exactly behavior that the app should have to be absolutely perfect. +10000 to this

**Comment by Lohoris** [comment]
I agree. This is what Jumpcut did. When I searched for a replacement, I ended up trying Clipy, but I had to abandon it because it lacked this behaviour, sorry.

**Comment by gsosna** [comment]
I agree also... furthermore, 90% of cases, I want to paste the 2nd item in the history. So the initial focus should be on the 2nd item IMHO.

**Comment by nj-io** [comment]
I've opened a PR covering the most common case discussed here: a global shortcut that pastes the 2nd most recent item directly, with no menu. Since Clipy re-records each paste as the newest entry, pressing it again alternates between the two most recent clips, which gives the "toggle between the last two things I copied" behaviour without a hold-and-cycle mode.

It isn't the full hold-modifier cycling from the original post. The paste path is index-based, so extending it to further positions is small, but proper cycling would also need an in-memory cursor rather than a database read, because the pasteboard is only polled every 500ms.

PR: https://github.com/Clipy/Clipy/pull/716
